torch/testing/_internal/common_utils.py

```diff
@@ -817,17 +819,44 @@ def settings(*args, **kwargs):
     print('Fail to import hypothesis in common_utils, tests are not derandomized')
 
 
+def fetch_and_cache(name: str, url: str) -> str:
+    """
+    Some tests run in a different process so globals like `slow_test_dict` won't
+    always be filled even though the test file was already downloaded on this
+    machine, so cache it on disk
+    """
+    path = os.path.join(tempfile.gettempdir(), name)
+
+    def is_cached_file_valid():
+        # Check if the file is new enough (say 1 hour for now). A real check
+        # could make a HEAD request and check/store the file's ETag
+        fname = pathlib.Path(path)
+        now = datetime.datetime.now()
+        mtime = datetime.datetime.fromtimestamp(fname.stat().st_mtime)
+        diff = now - mtime
+        return diff.total_seconds() < datetime.timedelta(hours=1).seconds
```

Comment:
Maybe do 3 hours as most tests run longer than 1 hour? And it may be a good idea to store the number of hours in a global variable like `FILE_CACHE_LIFESPAN` for more visibility and clarity.